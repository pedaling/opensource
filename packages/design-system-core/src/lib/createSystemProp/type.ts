import type { Theme } from '../types';

export type InterpolationFn = (props: Record<string, any>) => Record<string, any>;

export type SystemPropConfig = {
  property: string;
  styleProperty?: string;
  scale?: keyof Pick<Theme, 'colors' | 'space' | 'opacity'>;
  disabled?: boolean;
  transform?: (value: any) => Record<string, any> | undefined;
};

export type SystemProp = {
  (props: Record<string, any>): Record<string, any>;
  propName: string;
  disabled: boolean;
};
