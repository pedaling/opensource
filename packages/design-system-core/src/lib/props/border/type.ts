import type { ResponsiveValue } from '../../types';

export type BorderProps = {
  borderWidth?: ResponsiveValue<number>;
  borderStyle?: ResponsiveValue<'solid' | 'none'>;
  borderColor?: ResponsiveValue<string>;
  borderTopWidth?: ResponsiveValue<number>;
  borderTopStyle?: ResponsiveValue<'solid' | 'none'>;
  borderTopColor?: ResponsiveValue<string>;
  borderRightWidth?: ResponsiveValue<number>;
  borderRightStyle?: ResponsiveValue<'solid' | 'none'>;
  borderRightColor?: ResponsiveValue<string>;
  borderBottomWidth?: ResponsiveValue<number>;
  borderBottomStyle?: ResponsiveValue<'solid' | 'none'>;
  borderBottomColor?: ResponsiveValue<string>;
  borderLeftWidth?: ResponsiveValue<number>;
  borderLeftStyle?: ResponsiveValue<'solid' | 'none'>;
  borderLeftColor?: ResponsiveValue<string>;
  borderRadius?: ResponsiveValue<number | string>;
  borderTopLeftRadius?: ResponsiveValue<number | string>;
  borderTopRightRadius?: ResponsiveValue<number | string>;
  borderBottomLeftRadius?: ResponsiveValue<number | string>;
  borderBottomRightRadius?: ResponsiveValue<number | string>;
  outlineWidth?: ResponsiveValue<number>;
  outlineStyle?: ResponsiveValue<'solid' | 'none'>;
  outlineColor?: ResponsiveValue<string>;
  outlineOffset?: ResponsiveValue<number>;
};
