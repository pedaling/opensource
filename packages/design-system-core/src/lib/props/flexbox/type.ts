import type { ResponsiveValue } from '../../types';

export type FlexboxProps = {
  flex?: ResponsiveValue<string | number>;
  flexGrow?: ResponsiveValue<string | number>;
  flexShrink?: ResponsiveValue<string | number>;
  flexBasis?: ResponsiveValue<string | number>;
  flexDirection?: ResponsiveValue<'column' | 'column-reverse' | 'row' | 'row-reverse'>;
  flexWrap?: ResponsiveValue<'nowrap' | 'wrap' | 'wrap-reverse'>;
  alignContent?: ResponsiveValue<
    | 'space-around'
    | 'space-between'
    | 'space-evenly'
    | 'stretch'
    | 'center'
    | 'end'
    | 'flex-end'
    | 'flex-start'
    | 'start'
    | 'baseline'
    | 'normal'
  >;
  alignItems?: ResponsiveValue<
    | 'center'
    | 'end'
    | 'flex-end'
    | 'flex-start'
    | 'self-end'
    | 'self-start'
    | 'start'
    | 'baseline'
    | 'normal'
    | 'stretch'
  >;
  alignSelf?: ResponsiveValue<
    | 'center'
    | 'end'
    | 'flex-end'
    | 'flex-start'
    | 'self-end'
    | 'self-start'
    | 'start'
    | 'auto'
    | 'baseline'
    | 'normal'
    | 'stretch'
  >;
  justifyContent?: ResponsiveValue<
    | 'space-around'
    | 'space-between'
    | 'space-evenly'
    | 'stretch'
    | 'center'
    | 'end'
    | 'flex-end'
    | 'flex-start'
    | 'start'
    | 'left'
    | 'normal'
    | 'right'
  >;
  justifyItems?: ResponsiveValue<
    | 'center'
    | 'end'
    | 'flex-end'
    | 'flex-start'
    | 'self-end'
    | 'self-start'
    | 'start'
    | 'baseline'
    | 'left'
    | 'legacy'
    | 'normal'
    | 'right'
    | 'stretch'
  >;
  justifySelf?: ResponsiveValue<
    | 'center'
    | 'end'
    | 'flex-end'
    | 'flex-start'
    | 'self-end'
    | 'self-start'
    | 'start'
    | 'auto'
    | 'baseline'
    | 'left'
    | 'normal'
    | 'right'
    | 'stretch'
  >;
  order?: ResponsiveValue<number>;
};
