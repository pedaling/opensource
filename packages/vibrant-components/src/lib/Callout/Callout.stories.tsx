import type { ComponentMeta, ComponentStory } from '@storybook/react';
import { VStack } from '../VStack';
import { Callout } from './Callout';

export default {
  title: 'Callout',
  component: Callout,
  args: {
    title: '타이틀을 입력하세요',
    kind: 'warning',
    buttonText: 'Action',
    onButtonClick: () => {},
  },
} as ComponentMeta<typeof Callout>;

export const Basic: ComponentStory<typeof Callout> = props => (
  <VStack width="80%" p={10}>
    <Callout {...props} />
  </VStack>
);
