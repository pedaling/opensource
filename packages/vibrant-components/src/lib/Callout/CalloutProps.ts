import { propVariant, withVariation } from '@vibrant-ui/core';
import { Icon } from '@vibrant-ui/icons';

export type CalloutType = 'default' | 'information' | 'notice' | 'success' | 'warning';

export type CalloutProps = {
  title: string;
  description?: string;
  type?: CalloutType;
} & (
  | {
      action: string;
      onActionClick?: () => void;
    }
  | {
      action?: never;
      onActionClick?: never;
    }
);

export const withCalloutVariation = withVariation<CalloutProps>('Callout')(
  propVariant({
    props: [
      {
        name: 'type',
        keep: false,
        default: 'default',
      },
    ],
    variants: {
      default: {
        backgroundColor: 'surface2' as const,
        fontColor: 'onView1' as const,
        IconComponent: Icon.InfoCircle,
      },
      information: {
        backgroundColor: 'informativeContainer' as const,
        fontColor: 'onViewInformative' as const,
        IconComponent: Icon.InfoCircle,
      },
      notice: {
        backgroundColor: 'errorContainer' as const,
        fontColor: 'onViewError' as const,
        IconComponent: Icon.InfoCircle,
      },
      warning: {
        backgroundColor: 'warningContainer' as const,
        fontColor: 'onViewWarning' as const,
        IconComponent: Icon.Alert,
      },
      success: {
        backgroundColor: 'successContainer' as const,
        fontColor: 'onViewSuccess' as const,
        IconComponent: Icon.CheckCircle,
      },
    },
  })
);
