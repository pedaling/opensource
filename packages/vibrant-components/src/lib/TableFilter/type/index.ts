export type StringFilterOperator = 'contains' | 'empty' | 'equals' | 'notContains' | 'notEmpty' | 'notEquals';

export type StringFilter = {
  value: string;
  operator: StringFilterOperator[];
};

export type Filter = StringFilter & { dataKey: string };
