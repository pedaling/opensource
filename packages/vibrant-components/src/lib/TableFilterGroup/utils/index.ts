import type { DateFilterOperator, MultiSelectFilterOperator, Option, StringFilterOperator } from '../types';

export const isValueRequiredOperator = (op: string) => op === 'empty' || op === 'notEmpty';

export const isStringFilterValid = (filter: { value: string; operator: StringFilterOperator }) =>
  Boolean(filter.value) || isValueRequiredOperator(filter.operator);

export const isDateFilterValid = (filter: { value: Date[]; operator: DateFilterOperator }) => {
  if (isValueRequiredOperator(filter.operator)) return true;

  if (filter.operator === 'between') return filter.value.length >= 2;

  return filter.value.length >= 1;
};

export const isMultiSelectFilterValid = (filter: { value: Option['value'][]; operator: MultiSelectFilterOperator }) =>
  Boolean(filter.value.length > 0) || isValueRequiredOperator(filter.operator);
