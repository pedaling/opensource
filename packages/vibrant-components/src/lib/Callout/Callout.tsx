import { HStack, Pressable, Title, VStack } from '@vibrant-ui/components';
import { Box, Text } from '@vibrant-ui/core';
import { withCalloutVariation } from './CalloutProps';

export const Callout = withCalloutVariation(
  ({ title, description, buttonText, onButtonClick, backgroundColor, fontColor, IconComponent }) => (
    <Box
      width="100%"
      borderStyle="solid"
      backgroundColor={backgroundColor}
      borderWidth={1}
      borderColor="outline1"
      borderRadius={2}
    >
      <VStack p={16}>
        <HStack spacing={6}>
          <Box mt={2}>
            <IconComponent.Fill fill={fontColor} size={16} />
          </Box>
          <Title level={6} weight="bold" color={fontColor} overflowWrap="anywhere">
            {title}
          </Title>
        </HStack>
        {description ? (
          <Text
            mt={8}
            mb={description ? 2 : 0}
            lineHeight={18}
            fontSize={14}
            fontWeight="regular"
            overflowWrap="anywhere"
          >
            {description}
          </Text>
        ) : null}
        {buttonText ? (
          <Box mt={12} alignSelf="flex-end">
            <Pressable
              onClick={onButtonClick}
              interactions={['focus', 'active']}
              backgroundColor="surface1"
              borderRadius={2}
              overlayColor="onView1"
              py={10}
              px={14}
            >
              <Text lineHeight={18} fontSize={14} fontWeight="bold" color="onView1">
                {buttonText}
              </Text>
            </Pressable>
          </Box>
        ) : null}
      </VStack>
    </Box>
  )
);
