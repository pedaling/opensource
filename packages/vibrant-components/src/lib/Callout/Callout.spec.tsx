import { fireEvent } from '@testing-library/react';
import { ThemeProvider } from '@vibrant-ui/core';
import { createReactRenderer } from '@vibrant-ui/utils/testing-web';
import { Callout } from './Callout';
import type { CalloutProps, CalloutType } from './CalloutProps';

describe('<Callout />', () => {
  const { render } = createReactRenderer();

  let mockOnButtonClick: jest.Mock;

  beforeEach(() => {
    mockOnButtonClick = jest.fn();
  });

  const renderCallout = (props: Partial<CalloutProps> = {}) => {
    const calloutProps: any = {
      title: props.title,
      kind: props.kind,
      testId: props.testId || 'callout',
    };

    // Handle content props - must choose one or the other
    if (props.contents) {
      calloutProps.contents = props.contents;
    } else if (props.renderContents) {
      calloutProps.renderContents = props.renderContents;
    }

    // Handle button props conditionally
    if (props.buttonText) {
      calloutProps.buttonText = props.buttonText;
      calloutProps.onButtonClick = props.onButtonClick;
    }

    return render(
      <ThemeProvider theme={{ mode: 'light' }}>
        <Callout {...calloutProps} />
      </ThemeProvider>
    );
  };

  describe('Basic rendering', () => {
    it('renders with default props', () => {
      const renderer = renderCallout();

      expect(renderer.getByTestId('callout')).toBeTruthy();
    });

    it('renders with custom testId', () => {
      const renderer = renderCallout({ testId: 'custom-callout' });

      expect(renderer.getByTestId('custom-callout')).toBeTruthy();
    });
  });

  describe('Content rendering', () => {
    it('renders title when provided', () => {
      const title = 'Test Title';
      const renderer = renderCallout({ title });

      expect(renderer.getByText(title)).toBeTruthy();
    });

    it('renders content when provided', () => {
      const contents = 'Test content message';
      const renderer = renderCallout({ contents });

      expect(renderer.getByText(contents)).toBeTruthy();
    });

    it('renders both title and content when provided', () => {
      const title = 'Test Title';
      const contents = 'Test content message';
      const renderer = renderCallout({ title, contents });

      expect(renderer.getByText(title)).toBeTruthy();
      expect(renderer.getByText(contents)).toBeTruthy();
    });

    it('renders custom content with renderContents', () => {
      const customContent = 'Custom rendered content';
      const renderer = renderCallout({
        renderContents: () => <div data-testid="custom-content">{customContent}</div>,
      });

      const customElements = renderer.getAllByTestId('custom-content');

      expect(customElements.length).toBeGreaterThan(0);
      const textElements = renderer.getAllByText(customContent);

      expect(textElements.length).toBeGreaterThan(0);
    });
  });

  describe('Button behavior', () => {
    it('renders button when buttonText is provided', () => {
      const buttonText = 'Click Me';
      const renderer = renderCallout({ buttonText });

      expect(renderer.getByText(buttonText)).toBeTruthy();
    });

    it('does not render button when buttonText is not provided', () => {
      const renderer = renderCallout({ contents: 'Test content' });

      expect(renderer.queryByRole('button')).toBeNull();
    });

    it('calls onButtonClick when button is clicked', () => {
      const buttonText = 'Click Me';
      const renderer = renderCallout({
        buttonText,
        onButtonClick: mockOnButtonClick,
      });

      fireEvent.click(renderer.getByText(buttonText));
      expect(mockOnButtonClick).toHaveBeenCalledTimes(1);
    });
  });

  describe('Styling variations', () => {
    const calloutTypes: CalloutType[] = ['default', 'error', 'informative', 'success', 'warning'];

    it.each(calloutTypes)('renders with correct styling for %s kind', kind => {
      const renderer = renderCallout({
        kind,
        title: `${kind} Callout`,
        contents: `This is a ${kind} callout message`,
      });

      expect(renderer.getByTestId('callout')).toBeTruthy();
    });
  });

  describe('Box styling', () => {
    it('has the correct border radius', () => {
      const renderer = renderCallout({ contents: 'Test content' });

      expect(renderer.getByTestId('callout')).toHaveStyleRule('border-radius', '8px');
    });

    it('has the correct padding', () => {
      const renderer = renderCallout({ contents: 'Test content' });
      const element = renderer.getByTestId('callout');

      expect(element).toHaveStyleRule('padding-right', '14px');
      expect(element).toHaveStyleRule('padding-left', '16px');
      expect(element).toHaveStyleRule('padding-top', '8px');
      expect(element).toHaveStyleRule('padding-bottom', '8px');
    });
  });

  it('matches snapshot', () => {
    const renderer = renderCallout({
      title: 'Test Callout',
      contents: 'This is a test callout message',
      buttonText: 'Click Me',
      onButtonClick: mockOnButtonClick,
    });

    expect(renderer.container).toMatchSnapshot();
  });
});
