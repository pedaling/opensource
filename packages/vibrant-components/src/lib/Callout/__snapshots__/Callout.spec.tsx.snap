// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Callout /> matches snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 100%;
  border-style: solid;
  background-color: #ffffff;
  border-width: 1px;
  border-color: #0000000d;
  border-radius: 8px;
  padding-right: 14px;
  padding-left: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  box-sizing: border-box;
  position: relative;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  padding-top: 8px;
  padding-bottom: 8px;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 14px;
  height: 14px;
  fill: #0c0c0c;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  width: 12px;
  height: 0;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  padding-top: 6px;
  padding-bottom: 6px;
  width: 100%;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.emotion-7 {
  text-align: left;
  color: #0c0c0c;
  -webkit-text-fill-color: #0c0c0c;
  text-fill-color: #0c0c0c;
  overflow-wrap: anywhere;
  font-size: 0.875rem;
  line-height: 1.125rem;
  font-weight: 500;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  width: 0;
  height: 8px;
}

.emotion-9 {
  text-align: left;
  color: #0c0c0c;
  -webkit-text-fill-color: #0c0c0c;
  text-fill-color: #0c0c0c;
  overflow-wrap: anywhere;
  font-size: 0.875rem;
  line-height: 1.25rem;
  font-weight: 400;
}

.emotion-11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  cursor: pointer;
  background-color: #00000008;
  border-width: 0;
  padding: 0;
  overflow: hidden;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  z-index: 0;
  width: auto;
  padding-top: 7px;
  padding-bottom: 7px;
  padding-left: 8px;
  padding-right: 8px;
  border-radius: 4px;
}

.emotion-13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  box-sizing: border-box;
  position: relative;
  opacity: 1;
  gap: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-14 {
  text-align: left;
  color: #0c0c0c;
  -webkit-text-fill-color: #0c0c0c;
  text-fill-color: #0c0c0c;
  font-size: 0.75rem;
  line-height: 1rem;
  font-weight: 500;
  margin-left: 4px;
  margin-right: 4px;
}

.emotion-15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: absolute;
  left: 0;
  right: 0;
  top: 0;
  bottom: 0;
  background-color: #0c0c0c;
  pointer-events: none;
}

<div>
  <div
    class="emotion-0"
    data-testid="callout"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <svg
          class="emotion-3"
          data-testid="infocircle-fill"
          viewBox="0 0 24 24"
        >
          <path
            class="emotion-4"
            d="M12 1C5.9 1 1 5.9 1 12s4.9 11 11 11 11-4.9 11-11S18.1 1 12 1m1.25 16.4c0 .05-.05.1-.1.1h-2.3c-.05 0-.1-.05-.1-.1v-6.8c0-.05.05-.1.1-.1h2.3c.05 0 .1.05.1.1zM12 9c-.85 0-1.5-.65-1.5-1.5S11.15 6 12 6s1.5.65 1.5 1.5S12.85 9 12 9"
          />
        </svg>
      </div>
      <span
        class="emotion-5"
      />
      <div
        class="emotion-6"
      >
        <h3
          class="emotion-7"
          data-testid="title"
        >
          Test Callout
        </h3>
        <span
          class="emotion-8"
        />
        <span
          class="emotion-9"
          data-testid="paragraph"
        >
          This is a test callout message
        </span>
        <span
          class="emotion-8"
        />
        <div
          class="emotion-11"
        >
          <button
            class="emotion-12"
            data-testid="contained-button"
            style="opacity: 1; transition-property: opacity; transition-duration: 200ms; transition-timing-function: cubic-bezier(0.5, 1, 0.89, 1);"
            type="button"
          >
            <span
              class="emotion-13"
            >
              <span
                class="emotion-14"
                data-testid="button-text"
              >
                Click Me
              </span>
            </span>
            <span
              class="emotion-15"
              style="opacity: 0; transition-property: opacity; transition-duration: 200ms; transition-timing-function: cubic-bezier(0.5, 1, 0.89, 1);"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
